File: src/main/java/uknowlab/designpattern/factory/SimplePizzaFactory.java
package uknowlab.designpattern.factory;

import uknowlab.designpattern.factory.pizza.CheesePizza;
import uknowlab.designpattern.factory.pizza.PepperoniPizza;
import uknowlab.designpattern.factory.pizza.Pizza;
import uknowlab.designpattern.factory.pizza.VeggiePizza;

public class SimplePizzaFactory {

    public Pizza createPizza(String type) {
        return switch (type) {
            case "cheese" -> new CheesePizza();
            case "veggie" -> new VeggiePizza();
            case "pepperoni" -> new PepperoniPizza();
            default -> null;
        };
    }
}

File: src/main/java/uknowlab/designpattern4kotlin/factory/SimplePizzaFactory.kt
package uknowlab.designpattern4kotlin.factory

import uknowlab.designpattern4kotlin.factory.pizza.CheesePizza
import uknowlab.designpattern4kotlin.factory.pizza.PepperoniPizza
import uknowlab.designpattern4kotlin.factory.pizza.Pizza
import uknowlab.designpattern4kotlin.factory.pizza.VeggiePizza


class SimplePizzaFactory {
    fun createPizza(type: String?): Pizza {
        return when (type) {
            "cheese" -> CheesePizza()
            "veggie" -> VeggiePizza()
            "pepperoni" -> PepperoniPizza()
            else -> throw IllegalArgumentException("No such pizza.")
        }
    }
}